sys/dev/ath/if_ath_usb.c

```diff
@@ -434,13 +434,17 @@ static int
 ath_usb_attach(device_t self)
 {
 	struct usb_attach_arg *uaa = device_get_ivars(self);
-	struct ath_usb_softc *usc =  device_get_softc(self);
-	struct ath_softc *sc = &usc->sc_sc;
+	struct ath_softc *sc = device_get_softc(self);
 	struct ieee80211com *ic = &sc->sc_ic;
+	struct ath_usb_softc *usc;
 	usb_error_t error;
 	struct usb_endpoint *ep, *ep_end;
 
 	device_set_usb_desc(self);
+	usc = malloc(sizeof(struct ath_usb_softc), M_TEMP,
```

Comment:
Is the memory freed? 

I also wonder if there is a memory leak if we replace usc allocated by device_get by other structure, but let's think about it other time :)

Reply:
It is freed now and also I changed the size of structure 
`static driver_t ath_usb_driver = {
	.name = "if_ath_usb",
	.methods = ath_usb_methods,
	.size = sizeof(struct ath_softc)
};`